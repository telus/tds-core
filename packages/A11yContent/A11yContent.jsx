import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import { safeRest } from '@tds/util-helpers'
import { componentWithName, or } from '@tds/util-prop-types'

const StyledA11yContent = styled.span({
  position: 'absolute',
  height: '1px',
  width: '1px',
  overflow: 'hidden',
  clip: 'rect(1px, 1px, 1px, 1px)',
})

/**
 * @version ./package.json
 */

const A11yContent = ({ children, ...rest }) => {
  return <StyledA11yContent {...safeRest(rest)}>{children}</StyledA11yContent>
}

A11yContent.propTypes = {
  /**
   * Accessible content. Can be either a string or a `Heading` Component
   */
  children: or([PropTypes.string, componentWithName('Heading')]).isRequired,
}
A11yContent.defaultProps = {}

export default A11yContent
