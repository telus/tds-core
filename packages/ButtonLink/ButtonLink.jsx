import React, { forwardRef } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import { componentWithName, or } from '@tds/util-prop-types'
import { StyledButton } from '@tds/core-button'
import { links } from '@tds/shared-styles'
import {
  colorWhite as buttonTextColor,
  colorPrimary as primaryBgColor,
  colorSecondary as secondaryBgColor,
  colorText,
} from '@tds/core-colours'
import { safeRest } from '@tds/util-helpers'

import { warn } from '../../shared/utils/warn'

const StyledButtonLink = styled(StyledButton)(
  links.focusOutline,
  {
    textDecoration: 'none',
  },
  ({ variant }) => {
    let color
    let hoverColor
    if (variant === 'primary') {
      color = buttonTextColor
      hoverColor = primaryBgColor
    } else if (variant === 'secondary') {
      color = buttonTextColor
      hoverColor = secondaryBgColor
    } else {
      color = colorText
      hoverColor = buttonTextColor
    }

    return {
      '&:link,&:visited': {
        color,
      },
      '&:hover': {
        color: hoverColor,
      },
    }
  },
  ({ fullwidth }) => {
    let width
    if (fullwidth) {
      width = '100%'
    }
    return {
      '&:link,&:visited': {
        width,
      },
    }
  }
)

/**
 * A link that is styled as a button.
 *
 * @version ./package.json
 */
const ButtonLink = forwardRef(
  ({ reactRouterLinkComponent, variant, fullwidth, children, ...rest }, ref) => {
    if ((reactRouterLinkComponent || rest.to) && !(reactRouterLinkComponent && rest.to)) {
      warn('Link Button', 'The props `reactRouterLinkComponent` and `to` must be used together.')
    }

    return (
      <StyledButtonLink
        {...safeRest(rest)}
        as={reactRouterLinkComponent || 'a'}
        variant={variant}
        ref={ref}
        fullwidth={fullwidth}
      >
        {children}
      </StyledButtonLink>
    )
  }
)

ButtonLink.displayName = 'ButtonLink'

ButtonLink.propTypes = {
  /**
   * The style.
   */
  variant: PropTypes.oneOf(['primary', 'secondary', 'inverted']),
  /**
   * React Router Link component.
   */
  reactRouterLinkComponent: PropTypes.oneOfType([PropTypes.func, PropTypes.object]),
  /**
   * Target URL (if using 'reactRouterLinkComponent')
   */
  to: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
  /**
   * Target URL
   */
  href: PropTypes.string,
  /**
   * If `true`, sets `ButtonLink` to 100% width.
   * @since 2.2.0
   */
  fullwidth: PropTypes.bool,
  /**
   * The label. It can include the `A11yContent` component or strings.
   */
  children: or([PropTypes.string, componentWithName('A11yContent')]).isRequired,
}
ButtonLink.defaultProps = {
  variant: 'primary',
  reactRouterLinkComponent: null,
  to: null,
  href: null,
  fullwidth: false,
}

export default ButtonLink
