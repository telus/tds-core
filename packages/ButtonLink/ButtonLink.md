### Minimal usage

A Button Link is a navigational element that styles itself as a button. Clicking one should navigate to a new "page".

By default, Buttons will be displayed in the `primary` variant. Use primary Button Links for the main call to action on a page.

```jsx
<ButtonLink href="#">Find out how</ButtonLink>
```

### Usage criteria

- Use Buttons Links to navigate to a new "page"
- Aim to use only one button per page
- Keep the text short and able to fit on a single line (our recommendation is 2-24 characters)
- Make sure the button text describes an action
- Buttons should not be disabled

### Placing buttons on dark backgrounds

Use the `inverted` button link on top of a dark background (TELUS approved colours or images).

Use this variant with caution. There will be accessibility issues if the colour contrast of the image and the button text is too low in the hover state.

```jsx { "props": { "className": "docs_hero" }}
<ButtonLink href="#" variant="inverted">
  Advanced solutions
</ButtonLink>
```

### Using A11yContent

Use the `A11yContent` component to create invisible text that is read out loud by screen readers.

```jsx
<Button href="#">
  <span>
    Shop <A11yContent>iPhone</A11yContent>Now
  </span>
</Button>
```

### Full width option

The full width button option is available to be used in Cards where the button should fill the full width of the Card. Please use this only where it makes sense visually to do so.
