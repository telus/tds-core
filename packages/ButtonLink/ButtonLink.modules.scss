@import '~@tds/core-colours/colours';

$button-text-color: $color-white;
$primary-bg-color: $color-primary;
$secondary-bg-color: $color-secondary;

.base {
  composes: focusOutline from '../../shared/styles/Links.modules.scss';

  text-decoration: none;
}

.primary {
  composes: base;

  &:link,
  &:visited {
    color: $button-text-color;
  }

  &:hover {
    color: $primary-bg-color;
  }
}

.secondary {
  composes: base;

  &:link,
  &:visited {
    color: $button-text-color;
  }

  &:hover {
    color: $secondary-bg-color;
  }
}

.inverted {
  composes: base;

  &:link,
  &:visited {
    color: $color-text;
  }

  &:hover {
    color: $button-text-color;
  }
}
