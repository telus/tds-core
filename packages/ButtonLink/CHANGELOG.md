# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [2.2.0](https://github.com/telusdigital/tds/compare/@tds/core-button-link@2.1.15...@tds/core-button-link@2.2.0) (2020-07-22)


### Features

* **core-button-link:** add span as a valid child prop-type ([8507eb6](https://github.com/telusdigital/tds/commit/8507eb60ed12e39652ed29004bbfd4092f795492))





## [2.1.15](https://github.com/telusdigital/tds/compare/@tds/core-button-link@2.1.14...@tds/core-button-link@2.1.15) (2020-03-25)


### Bug Fixes

* **core-button-link:** prop type warnings for reactRouterLinkComponent ([d30d356](https://github.com/telusdigital/tds/commit/d30d35657f018e45a2746066f3223d1d4d9b8db1)), closes [#1401](https://github.com/telusdigital/tds/issues/1401)





## [2.1.14](https://github.com/telusdigital/tds/compare/@tds/core-button-link@2.1.13...@tds/core-button-link@2.1.14) (2020-02-03)

**Note:** Version bump only for package @tds/core-button-link





## [2.1.13](https://github.com/telusdigital/tds/compare/@tds/core-button-link@2.1.12...@tds/core-button-link@2.1.13) (2020-01-27)

**Note:** Version bump only for package @tds/core-button-link





## [2.1.12](https://github.com/telusdigital/tds/compare/@tds/core-button-link@2.1.11...@tds/core-button-link@2.1.12) (2020-01-23)

**Note:** Version bump only for package @tds/core-button-link





## [2.1.11](https://github.com/telusdigital/tds/compare/@tds/core-button-link@2.1.10...@tds/core-button-link@2.1.11) (2020-01-22)

**Note:** Version bump only for package @tds/core-button-link





## [2.1.10](https://github.com/telusdigital/tds/compare/@tds/core-button-link@2.1.9...@tds/core-button-link@2.1.10) (2019-10-09)

**Note:** Version bump only for package @tds/core-button-link





## [2.1.9](https://github.com/telusdigital/tds/compare/@tds/core-button-link@2.1.8...@tds/core-button-link@2.1.9) (2019-10-04)

**Note:** Version bump only for package @tds/core-button-link





## [2.1.8](https://github.com/telusdigital/tds/compare/@tds/core-button-link@2.1.7...@tds/core-button-link@2.1.8) (2019-09-12)

**Note:** Version bump only for package @tds/core-button-link





## [2.1.7](https://github.com/telusdigital/tds/compare/@tds/core-button-link@2.1.6...@tds/core-button-link@2.1.7) (2019-09-06)

**Note:** Version bump only for package @tds/core-button-link





## [2.1.6](https://github.com/telusdigital/tds/compare/@tds/core-button-link@2.1.5...@tds/core-button-link@2.1.6) (2019-08-22)

**Note:** Version bump only for package @tds/core-button-link





## [2.1.5](https://github.com/telusdigital/tds/compare/@tds/core-button-link@2.1.4...@tds/core-button-link@2.1.5) (2019-08-09)

**Note:** Version bump only for package @tds/core-button-link





## [2.1.4](https://github.com/telusdigital/tds/compare/@tds/core-button-link@2.1.3...@tds/core-button-link@2.1.4) (2019-08-01)

**Note:** Version bump only for package @tds/core-button-link





## [2.1.3](https://github.com/telusdigital/tds/compare/@tds/core-button-link@2.1.2...@tds/core-button-link@2.1.3) (2019-07-30)

**Note:** Version bump only for package @tds/core-button-link





## [2.1.2](https://github.com/telusdigital/tds/compare/@tds/core-button-link@2.1.1...@tds/core-button-link@2.1.2) (2019-07-25)

**Note:** Version bump only for package @tds/core-button-link





## [2.1.1](https://github.com/telusdigital/tds/compare/@tds/core-button-link@2.1.0...@tds/core-button-link@2.1.1) (2019-06-25)

**Note:** Version bump only for package @tds/core-button-link





# [2.1.0](https://github.com/telusdigital/tds/compare/@tds/core-button-link@2.0.11...@tds/core-button-link@2.1.0) (2019-06-13)


### Features

* **core-button-link:** add forwardRef ([6c5383d](https://github.com/telusdigital/tds/commit/6c5383d))





## [2.0.11](https://github.com/telusdigital/tds/compare/@tds/core-button-link@2.0.10...@tds/core-button-link@2.0.11) (2019-06-06)

**Note:** Version bump only for package @tds/core-button-link





## [2.0.10](https://github.com/telusdigital/tds/compare/@tds/core-button-link@2.0.9...@tds/core-button-link@2.0.10) (2019-06-04)

**Note:** Version bump only for package @tds/core-button-link

## [2.0.9](https://github.com/telusdigital/tds/compare/@tds/core-button-link@2.0.8...@tds/core-button-link@2.0.9) (2019-05-14)

**Note:** Version bump only for package @tds/core-button-link

## [2.0.8](https://github.com/telusdigital/tds/compare/@tds/core-button-link@2.0.7...@tds/core-button-link@2.0.8) (2019-05-13)

**Note:** Version bump only for package @tds/core-button-link

## [2.0.7](https://github.com/telusdigital/tds/compare/@tds/core-button-link@2.0.6...@tds/core-button-link@2.0.7) (2019-05-07)

**Note:** Version bump only for package @tds/core-button-link

## [2.0.6](https://github.com/telusdigital/tds/compare/@tds/core-button-link@2.0.5...@tds/core-button-link@2.0.6) (2019-05-01)

**Note:** Version bump only for package @tds/core-button-link

## [2.0.5](https://github.com/telusdigital/tds/compare/@tds/core-button-link@2.0.4...@tds/core-button-link@2.0.5) (2019-05-01)

**Note:** Version bump only for package @tds/core-button-link

## [2.0.4](https://github.com/telusdigital/tds/compare/@tds/core-button-link@2.0.3...@tds/core-button-link@2.0.4) (2019-04-23)

**Note:** Version bump only for package @tds/core-button-link

## [2.0.3](https://github.com/telusdigital/tds/compare/@tds/core-button-link@2.0.2...@tds/core-button-link@2.0.3) (2019-04-22)

**Note:** Version bump only for package @tds/core-button-link

## [2.0.2](https://github.com/telusdigital/tds/compare/@tds/core-button-link@2.0.1...@tds/core-button-link@2.0.2) (2019-04-04)

**Note:** Version bump only for package @tds/core-button-link

## [2.0.1](https://github.com/telusdigital/tds/compare/@tds/core-button-link@2.0.0...@tds/core-button-link@2.0.1) (2019-03-28)

### Bug Fixes

- **core-button-link:** remove @tds/shared-base-button from deps ([08c17a1](https://github.com/telusdigital/tds/commit/08c17a1))

# [2.0.0](https://github.com/telusdigital/tds/compare/@tds/core-button-link@1.1.10...@tds/core-button-link@2.0.0) (2019-03-28)

### Features

- **core-button-link:** upgrade to styled-components ([1e9889d](https://github.com/telusdigital/tds/commit/1e9889d))

### BREAKING CHANGES

- **core-button-link:** upgrade to styled-components

## [1.1.10](https://github.com/telusdigital/tds/compare/@tds/core-button-link@1.1.9...@tds/core-button-link@1.1.10) (2019-03-28)

**Note:** Version bump only for package @tds/core-button-link

## [1.1.9](https://github.com/telusdigital/tds/compare/@tds/core-button-link@1.1.8...@tds/core-button-link@1.1.9) (2019-03-20)

**Note:** Version bump only for package @tds/core-button-link

## [1.1.8](https://github.com/telusdigital/tds/compare/@tds/core-button-link@1.1.7...@tds/core-button-link@1.1.8) (2019-03-20)

**Note:** Version bump only for package @tds/core-button-link

## [1.1.7](https://github.com/telusdigital/tds/compare/@tds/core-button-link@1.1.6...@tds/core-button-link@1.1.7) (2019-03-18)

**Note:** Version bump only for package @tds/core-button-link

## [1.1.6](https://github.com/telusdigital/tds/compare/@tds/core-button-link@1.1.5...@tds/core-button-link@1.1.6) (2019-03-11)

**Note:** Version bump only for package @tds/core-button-link

## [1.1.5](https://github.com/telusdigital/tds/compare/@tds/core-button-link@1.1.4...@tds/core-button-link@1.1.5) (2019-03-05)

**Note:** Version bump only for package @tds/core-button-link

## [1.1.4](https://github.com/telusdigital/tds/compare/@tds/core-button-link@1.1.3...@tds/core-button-link@1.1.4) (2019-02-06)

**Note:** Version bump only for package @tds/core-button-link

## [1.1.3](https://github.com/telusdigital/tds/compare/@tds/core-button-link@1.1.2...@tds/core-button-link@1.1.3) (2019-01-10)

### Bug Fixes

- **core-button-link:** replace airbnb proptypes with util-prop-types ([2dfed63](https://github.com/telusdigital/tds/commit/2dfed63))

<a name="1.1.2"></a>

## [1.1.2](https://github.com/telusdigital/tds/compare/@tds/core-button-link@1.1.1...@tds/core-button-link@1.1.2) (2018-12-19)

**Note:** Version bump only for package @tds/core-button-link

<a name="1.1.1"></a>

## [1.1.1](https://github.com/telusdigital/tds/compare/@tds/core-button-link@1.1.0...@tds/core-button-link@1.1.1) (2018-07-23)

**Note:** Version bump only for package @tds/core-button-link

<a name="1.1.0"></a>

# [1.1.0](https://github.com/telusdigital/tds/compare/@tds/core-button-link@1.0.2...@tds/core-button-link@1.1.0) (2018-07-17)

### Bug Fixes

- **shared-base-button:** add span wrapper to support spaces ([b9a1984](https://github.com/telusdigital/tds/commit/b9a1984))

### Features

- **core-button:** add A11yContent props ([d8d540d](https://github.com/telusdigital/tds/commit/d8d540d))
- **core-button-link:** add a11yContent props ([aae9310](https://github.com/telusdigital/tds/commit/aae9310))
- **core-button-link:** add A11yContent props ([7867a41](https://github.com/telusdigital/tds/commit/7867a41))
- **core-button-link:** support A11yContent as a child component ([cf9648b](https://github.com/telusdigital/tds/commit/cf9648b))

<a name="1.0.2"></a>

## [1.0.2](https://github.com/telusdigital/tds/compare/@tds/core-button-link@1.0.1...@tds/core-button-link@1.0.2) (2018-04-02)

### Bug Fixes

- **shared-base-button:** change sizing class name ([449e4b4](https://github.com/telusdigital/tds/commit/449e4b4))
- **shared-base-button:** refactor for sass based responsiveness ([a17f6ac](https://github.com/telusdigital/tds/commit/a17f6ac))

<a name="1.0.1"></a>

## [1.0.1](https://github.com/telusdigital/tds/compare/@tds/core-button-link@1.0.0...@tds/core-button-link@1.0.1) (2018-03-22)

### Bug Fixes

- **core-button-link:** standardize focus outline across browsers ([098cb03](https://github.com/telusdigital/tds/commit/098cb03)), closes [#445](https://github.com/telusdigital/tds/issues/445)
