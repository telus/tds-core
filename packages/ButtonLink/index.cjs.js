const ButtonLink = require('./dist/index.cjs')

module.exports = ButtonLink
