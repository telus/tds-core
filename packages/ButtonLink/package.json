{
  "name": "@tds/core-button-link",
  "version": "2.2.1",
  "description": "A link that is styled as a button",
  "main": "index.cjs.js",
  "module": "index.es.js",
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/telusdigital/tds.git"
  },
  "publishConfig": {
    "access": "public"
  },
  "author": "TELUS digital",
  "engines": {
    "node": ">=8"
  },
  "bugs": {
    "url": "https://github.com/telusdigital/tds/issues"
  },
  "homepage": "http://tds.telus.com",
  "peerDependencies": {
    "react": "^16.8.2",
    "react-dom": "^16.8.2",
    "styled-components": "^4.1.3"
  },
  "dependencies": {
    "@tds/core-button": "^2.2.0",
    "@tds/core-colours": "^2.2.1",
    "@tds/shared-styles": "^1.5.2",
    "@tds/util-helpers": "^1.4.2",
    "@tds/util-prop-types": "^1.4.0",
    "prop-types": "^15.5.10"
  },
  "devDependencies": {
    "@tds/core-a11y-content": "^2.1.5"
  }
}
