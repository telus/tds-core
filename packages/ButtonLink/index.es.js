import ButtonLink from './dist/index.es'

export default ButtonLink
