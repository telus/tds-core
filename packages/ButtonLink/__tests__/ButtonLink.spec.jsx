import React from 'react'
import { mount, shallow, render } from 'enzyme'
import A11yContent from '@tds/core-a11y-content'

import { warn } from '../../../shared/utils/warn'

import ButtonLink from '../ButtonLink'

jest.mock('../../../shared/utils/warn')

describe('ButtonLink', () => {
  const doMount = (overrides = {}) => {
    const link = mount(<ButtonLink {...overrides}>Go home</ButtonLink>)

    return link.find('a')
  }

  const doShallow = (overrides = {}) => {
    const link = shallow(<ButtonLink {...overrides}>Go home</ButtonLink>)

    return link
  }

  afterEach(() => {
    jest.clearAllMocks()
  })

  it('renders', () => {
    const link = render(<ButtonLink>Go home</ButtonLink>)

    expect(link).toMatchSnapshot()
  })

  it('is an anchor HTML element when using the href attribute', () => {
    const link = doMount({ href: 'http://telus.com' })

    expect(link).toHaveTagName('a')
    expect(link).toHaveProp('href', 'http://telus.com')
  })

  it('renders a react router link element when passed as a prop', () => {
    const MyLink = () => <span />
    const link = mount(<ButtonLink reactRouterLinkComponent={MyLink}>The link test</ButtonLink>)

    expect(link.find(MyLink)).toExist()
  })

  it('must use `reactRouterLinkComponent` and `to` props together', () => {
    const MyLink = () => <span />
    doShallow({ reactRouterLinkComponent: MyLink })

    expect(warn).toHaveBeenCalled()

    jest.clearAllMocks()

    const link = doShallow({ to: '/about' })

    expect(link).toHaveProp('to')
    expect(warn).toHaveBeenCalled()
  })

  it('can be presented as one of the allowed variants', () => {
    let link = doMount()
    expect(link).toHaveClassName('primary')

    link = doMount({ variant: 'secondary' })
    expect(link).toHaveClassName('secondary')
  })

  it('passes additional attributes to link element', () => {
    const link = doShallow({ id: 'the-link', tabIndex: 1 })

    expect(link).toHaveProp('id', 'the-link')
    expect(link).toHaveProp('tabIndex', 1)
  })

  describe('A11yContent', () => {
    it('connects to ButtonLink', () => {
      const link = shallow(
        <ButtonLink>
          Go home
          <A11yContent>testing</A11yContent>
        </ButtonLink>
      )

      expect(link).toContainReact(<A11yContent>testing</A11yContent>)
    })
  })

  it('does not allow custom CSS', () => {
    const link = doShallow({ className: 'my-custom-class', style: { color: 'hotpink' } })

    expect(link).not.toHaveProp('className', 'my-custom-class')
    expect(link).not.toHaveProp('style')
  })
})
