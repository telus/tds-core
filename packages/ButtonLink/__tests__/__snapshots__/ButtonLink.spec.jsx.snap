// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonLink can be presented as one of the allowed fullwidth 1`] = `
.c0 {
  border-width: 0;
  border-radius: 4px;
  word-wrap: break-word;
  font-size: 1rem;
  -webkit-letter-spacing: -0.8px;
  -moz-letter-spacing: -0.8px;
  -ms-letter-spacing: -0.8px;
  letter-spacing: -0.8px;
  line-height: 1.5;
  word-wrap: break-word;
  font-weight: 700;
  font-family: "TELUS-Web","Helvetica Neue",Helvetica,Arial,sans-serif;
  background-color: #2B8000;
  color: #fff;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover {
  box-shadow: 0 0 0 1px;
  background-color: #fff;
  color: #2B8000;
}

.c0:focus {
  outline: auto 5px -webkit-focus-ring-color;
}

.c0:link,
.c0:visited {
  color: #fff;
}

.c0:hover {
  color: #2B8000;
}

.c0:link,
.c0:visited {
  width: 100%;
}

<a
  class="c0"
>
  Go home
</a>
`;

exports[`ButtonLink can be presented as one of the allowed variants 1`] = `
.c0 {
  border-width: 0;
  border-radius: 4px;
  word-wrap: break-word;
  font-size: 1rem;
  -webkit-letter-spacing: -0.8px;
  -moz-letter-spacing: -0.8px;
  -ms-letter-spacing: -0.8px;
  letter-spacing: -0.8px;
  line-height: 1.5;
  word-wrap: break-word;
  font-weight: 700;
  font-family: "TELUS-Web","Helvetica Neue",Helvetica,Arial,sans-serif;
  background-color: #4b286d;
  color: #fff;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover {
  box-shadow: 0 0 0 1px;
  background-color: #fff;
  color: #4b286d;
}

.c0:focus {
  outline: auto 5px -webkit-focus-ring-color;
}

.c0:link,
.c0:visited {
  color: #fff;
}

.c0:hover {
  color: #4b286d;
}

.c0:link,
.c0:visited {
  width: auto;
}

<a
  class="c0"
>
  Go home
</a>
`;

exports[`ButtonLink renders 1`] = `
.c0 {
  border-width: 0;
  border-radius: 4px;
  word-wrap: break-word;
  font-size: 1rem;
  -webkit-letter-spacing: -0.8px;
  -moz-letter-spacing: -0.8px;
  -ms-letter-spacing: -0.8px;
  letter-spacing: -0.8px;
  line-height: 1.5;
  word-wrap: break-word;
  font-weight: 700;
  font-family: "TELUS-Web","Helvetica Neue",Helvetica,Arial,sans-serif;
  background-color: #2B8000;
  color: #fff;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover {
  box-shadow: 0 0 0 1px;
  background-color: #fff;
  color: #2B8000;
}

.c0:focus {
  outline: auto 5px -webkit-focus-ring-color;
}

.c0:link,
.c0:visited {
  color: #fff;
}

.c0:hover {
  color: #2B8000;
}

.c0:link,
.c0:visited {
  width: auto;
}

<a
  class="c0"
>
  Go home
</a>
`;
